### PR TITLE
Deny execute queries on disposed connections to avoid connection leaks

### PR DESCRIPTION
**Pull request by ili**
#445

This to fail in error immediately when wrong using pattern is applied

### CODE REVIEW

**Comment by MaceWindu** [comment]
Funny that couple of weeks we had this issue due to return of linq query from transaction scope and I even planned to implement protection from it in our application. But having it fixed in linq2db is even better.

For now new linq2db releases published without release notes. Maybe it makes sense to provide them with new releases? I expect to receive bug reports about broken applications when this fix approved and released. Good release notes will help users to understand what is going on.

**Comment by ili** [comment]
>For now new linq2db releases published without release notes. Maybe it makes sense to provide them with new releases? I expect to receive bug reports about broken applications when this fix approved and released. Good release notes will help users to understand what is going on.

Certainly you are right. We need some mechanism to build release notes from issues and pull requests... I think there is some suggested by GitHub, and, shame on me, I do not know it :(

File: Source/Linq/DataContextInfo.cs [hunk 1]
```diff
@@ -7,11 +7,11 @@ namespace LinqToDB.Linq
...
-		public DataContextInfo(IDataContext dataContext)
-		{
-			DataContext    = dataContext;
-			DisposeContext = false;
-		}
+		//public DataContextInfo(IDataContext dataContext)
+		//{
+		//	DataContext    = dataContext;
+		//	DisposeContext = false;
+		//}
```
Comment (sdanyliv) [inline Source/Linq/DataContextInfo.cs]: Why this constructor commented? Maybe just remove?

File: Source/SqlQuery/SelectQuery.cs [hunk 2]
```diff
@@ -168,31 +168,82 @@ public   string  Alias
...
+				//	|| new QueryVisitor().Find(Expression, e =>
+				//		{
+				//			switch (e.ElementType)
+				//			{
+				//				case QueryElementType.Column: return ((Column)e).Expression.Equals(other.Expression);
+				//			}
+				//			return false;
+				//		}) != null;
+
+				//return found;
```
Comment (sdanyliv) [inline Source/SqlQuery/SelectQuery.cs]: Commented for history?

**Comment by MaceWindu** [comment]
For release notes maybe we could look at those tools?
https://habrahabr.ru/company/eastbanctech/blog/323760/#cz

File: Source/ServiceModel/RemoteDataContextBase.cs [hunk 3]
```diff
@@ -353,6 +361,8 @@ IDataReader IDataContext.ExecuteReader(object query)
 
 		public void ReleaseQuery(object query)
 		{
+			ThrowOnDisposed();
```
Comment (sdanyliv) [inline Source/ServiceModel/RemoteDataContextBase.cs]: It is good place for throwing?

Comment (ili) [inline Source/ServiceModel/RemoteDataContextBase.cs]: The Idea was to throw exception on calling `IDataContext` methods (almost all methods).

**Pull request merged**